# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/HttpChannelChild.h
# Commit: 4214217c588c
# Full Hash: 4214217c588cd8f5a21acf197698905528f1ea17
# Author: Junior Hsu <juhsu@mozilla.com>
# Date: 2020-06-30 10:00:42
# Regressor Bug: 1633935
# File Overlap Count: 1
# Description:
#   Bug 1633935 - P5 allow OnProgress/OnStatus going thru pBg IPC, r=mayhemer,necko-reviewers
#   
#   Depends on D76971
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D76972
# ==============================================================================

diff -r e6c12eec1cef -r 4214217c588c netwerk/protocol/http/HttpChannelChild.h
--- a/netwerk/protocol/http/HttpChannelChild.h	Tue Jun 30 00:33:01 2020 +0000
+++ b/netwerk/protocol/http/HttpChannelChild.h	Tue Jun 30 00:33:19 2020 +0000
@@ -209,11 +209,6 @@
                                  const nsAString& aURL,
                                  const nsAString& aContentType) override;
 
-  mozilla::ipc::IPCResult RecvOnProgress(const int64_t& aProgress,
-                                         const int64_t& aProgressMax) override;
-
-  mozilla::ipc::IPCResult RecvOnStatus(const nsresult& aStatus) override;
-
  private:
   // We want to handle failure result of AsyncOpen, hence AsyncOpen calls the
   // Internal method
@@ -271,6 +266,9 @@
       const nsTArray<ConsoleReportCollected>& aConsoleReports);
   void ProcessFlushedForDiversion();
   void ProcessDivertMessages();
+  void ProcessOnProgress(const int64_t& aProgress, const int64_t& aProgressMax);
+
+  void ProcessOnStatus(const nsresult& aStatus);
 
   // Return true if we need to tell the parent the size of unreported received
   // data